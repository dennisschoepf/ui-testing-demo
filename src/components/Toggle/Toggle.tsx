import * as React from 'react';
import styled from 'styled-components';

const ToggleWrapper = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const ToggleDisplay = styled('div')<{ isVisible: boolean }>`
  margin-top: 2rem;
  visibility: ${props => (props.isVisible ? 'visible' : 'hidden')};
`;

const Button = styled.button`
  padding: 0.4rem 2rem 0.5rem;
  border: 2px solid white;
  background-color: #360028;
  color: white;
  font-weight: 700;
  font-size: 1.3rem;

  &:hover {
    cursor: pointer;
    background-color: white;
    color: #360028;
  }
`;

export default () => {
  const [isVisible, setIsVisible] = React.useState(false);

  return (
    <ToggleWrapper>
      <Button onClick={() => setIsVisible(!isVisible)}>Toggle</Button>
      <ToggleDisplay isVisible={isVisible}>123</ToggleDisplay>
    </ToggleWrapper>
  );
};
