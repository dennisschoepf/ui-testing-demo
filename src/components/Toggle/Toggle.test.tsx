import React from 'react';
import 'jest-styled-components';
import { mount } from 'enzyme';
import Toggle, { ToggleDisplay } from './Toggle';

describe('Component: Toggle', () => {
  /**
   * 1. Rendering & Visual Regression
   */
  it('should render', () => {
    const wrapper = mount(<Toggle />);
    expect(wrapper).toMatchSnapshot();
  });

  /**
   * 2. Interaktion
   */
  it('should toggle <ToggleDisplay />, after clicked', () => {
    const wrapper = mount(<Toggle />);
    expect(wrapper.find(ToggleDisplay).toHaveStyleRule('visibility', 'hidden'));
    // wrapper.simulate('click');
  });
});
