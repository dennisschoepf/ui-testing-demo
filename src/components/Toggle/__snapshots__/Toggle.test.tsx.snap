// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Toggle should render 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  margin-top: 2rem;
  visibility: hidden;
}

.c1 {
  padding: 0.4rem 2rem 0.5rem;
  border: 2px solid white;
  background-color: #360028;
  color: white;
  font-weight: 700;
  font-size: 1.3rem;
}

.c1:hover {
  cursor: pointer;
  background-color: white;
  color: #360028;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    Toggle
  </button>
  <div
    className="c2"
  >
    123
  </div>
</div>
`;
